### ISSUE TITLE
How to parse output from ONNX export

### ISSUE THREAD

**Issue by ghsanti** [Megvii-BaseDetection/YOLOX#1631]
I have finally trained the model and got decent results using YOLOX small. So running `demo.py` in my test set retrieves correct bboxes in most cases.

I get more than 8000 boxes:

```
  [
    -0.02134212851524353,
    1.0599262714385986,
    2.8227124214172363,
    1.3711082935333252,
    0.0000035861446576745948,
    0.2182561606168747
  ],
```

It seems the function [postprocess](https://github.com/Megvii-BaseDetection/YOLOX/blob/618fd8c08b2bc5fac9ffbb19a3b7e039ea0d5b9a/yolox/utils/boxes.py#LL32-L55) should be implemented, can you give some hints please? Is this correct?

Any ideas @FateScript ?

**Comment by mahnehsilla** [comment]
The steps I followed so far:

```
# convert my model
python3 tools/export_onnx.py --output-name yolox_s.onnx -f ../Downloads/yolox_s.py -c best_ckpt.pth 
# install onnx tensorflow
pip3 install onnx-tf==1.10
# install if the last command fails
pip3 install tensorflow-probability
onnx-tf convert -i yolox_s.onnx -o yolox_s.pb
```
I would still need to move this to a JS useful model. I wonder if there is a better way.

**Comment by mahnehsilla** [comment]
I think the way to go is to use onnx for inference, I still couldnt make it work

**Comment by FateScript** [comment]
@mahnehsilla ONNX output should be decoded to bounding box format.  For those users who are using YOLOX with onnx, plz add post processing code. Some references that might help you:  [c++  code](https://github.com/Megvii-BaseDetection/YOLOX/blob/main/demo/ncnn/cpp/yolox.cpp#L202-L247).  
For more code needed, check demo folder in YOLOX.

**Comment by mahnehsilla** [comment]
But that uses anchors? Would it be correct to do what you do in the `demo.py`?

**Comment by mahnehsilla** [comment]
Thank you. It's just that in JS most of those functions do not exist, but I will see what can be done.

**Comment by mahnehsilla** [comment]
This is closer, I think, and will help others in JS. 
I'd appreciate any help understanding the python code though. @FateScript

**Comment by mahnehsilla** [comment]
Why is the `postprocess` from `demo.py` different to the post processing that should be applied in onnx ? @FateScript

**Comment by mahnehsilla** [comment]
Finally managed to get the results in NodeJS (using Typescript) ! 
@FateScript 

I will upload it in MIT as well, uses TFJS and ONNX, and can run in browser.

**Comment by FateScript** [comment]
Excellent work @mahnehsilla : )

**Comment by hhaidar** [comment]
@ghsanti would you mind posting your code?

**Comment by ghsanti** [comment]
@hhaidar I wouldn't but I don't have access anymore. I can tell you though that it's just the pre/post processing steps written in TensorflowJS instead; ChatGPT can almost do it by itself.
